### DISCUSSION TITLE
Using Siri to stop playback from Music Assistant

### DISCUSSION THREAD

**Discussion by mayorgergich** [music-assistant/support#1561] [Q&A] [answered]
First of all, love this project. Been a game changer for many automations.

My home primarily uses HomePods and playback/control has been great with one glaring exception, I can't control my voice to stop playback to the HomePod as the HomePod doesn't even seem to realize it's playing music at all. When signaled using "Hey Siri, stop the music" it states nothing is playing and continues on playing the music on the speaker.

Does anyone have a workaround? I hate to have to manually disable playback every time.

**Comment by mayorgergich** [comment]
Buying this again. Anyone have insight?

This problem also extends to the Apple Home app, where the HomePod displays it is playing but offers no controls to Stop, Start, or control the volume while Music Assistant has control of the playback.

**Reply by OzGav** [reply]
I don’t use Apple devices but this is somewhat to be expected as the stream is being sent directly to the device. I have noted this and will speak to the devs in due course to see if anything can be done.

**Reply by curlycraze** [reply]
Within an hour of installing Music Assistant and being totally excited about it, I ran into this same issue. Like you, I heavily use the Home app or Siri to control Homepods. Not being able to stop the music via Home app or Siri feels like a dealbreaker :(  I did read the comment below about a workaround, but I have multiple Homepods and would have to teach all house occupants the various Siri commands to stop each one. Low approval factor. The ability to stop/start/control via Siri when Music Assistant is streaming to the Homepod would lead to widespread usage of this in my house :)

**Reply by JDogg016** [reply]
Maybe trying to resurrect this because being able to stop music in MA using Siri would be a game changer. 

Wouldn't this be possible if MA had a dedicated iOS app?

**Reply by Schlichi83** [reply]
This is also currently the dealbreaker for me. A solution would be GREATLY appreciated!

**Comment by freddeh** [comment, marked as answer]
You can builtd yourself a workaround with a switch/input boolean named ‘stop music’ or similar from Home Assistant made available through HomeKit. Then built an automation in Home Assistant to turn off music in Music Assistant when pressed. Siri can also trigger this command through voice.

**Reply by Schlichi83** [reply]
Yes, you can. But that is not a proper workaround, because you have to specifically use this custom voice command to stop the music. The regular commands you use for Siri to stop the music don't work. And while streaming to HomePod via Music Assistant the Home Pod disappears in the Music App on your iPhone - you don't see it anymore. In Home Assistant it's not showed as "playing" even though it plays. I assume that something is not properly configured with the AirPlay-streaming. Please fix this! Until then, Music Assistant is not really usable for users with HomePods and Apple Music.

**Reply by freddeh** [reply]
I know, but it’s the best for now…

**Comment by jamesbryant** [comment]
Echoing @mayorgergich — it's a game changer.

But I'll also add that not only do my HomePod hardware and Siri controls get rendered useless, the HomePod's presence across other devices disappears too. It would remove a few redundant speakers in my house if this supported AirPlay properly.

**Reply by Schlichi83** [reply]
You're right. This needs to be fixed.

**Comment by marcelveldt** [comment]
For now there is no way to fix this. Music Assistant's Airplay implementation is still based around lossless RAOP (Airplay 1) streaming and not based around AirPlay 2 which has a couple more features, especially for video. The AirPlay 2 protocol for audio should be more or less backwards compatible but in reality Apple doesn't give a crap and has dropped a lot of features, such as informing the internal systems that a stream is active, forward the metadata etc.

This is not fixable by us, until we migrate to AirPlay 2. A project that currently has no priority due to the fact the protocol is not fully reverse engineered yet and it will be a multi-month project with other higher priorities. We are a small team of volunteers, not some multi national company with many developers.

**Reply by Schlichi83** [reply]
Since AirPlay 2 is now implemented, can you maybe take another look at this? The problem still exists with all my HomePods with the new Music Assistant release. I switched my speakers to AirPlay 2 Protocol.

**Reply by JDogg016** [reply]
This would put the WAF factor over the top.

**Reply by marcelveldt** [reply]
We can not influence this, only Apple can

**Reply by timwnl** [reply]
> We can not influence this, only Apple can

Thanks for all your work on this. Airplay 2 was the reason I reinstalled MA, sad to see this comment. 

Just wondering if this is an Airplay 2 issue or Apple blocking any initiatives like MA to use this feature in Airplay 2? If so, what would be the additional benefit would be using Airplay 2 instead of Airplay 1? 

My challenge is to easy adjust audio volume when using MusicAssistant. In some situations I want to quickly / briefly lower the volume - any idea what would be the best solution not to use the iOS Control Center? I am aware of the HomeAssistant Switch workaround, but for volume control it would be a very complex setup (ability for Siri to understand, desired level of volume, desired duration of lowered volume etc.).
